plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/NewApplicationWizardModel.java

```diff
@@ -215,8 +221,12 @@ public boolean isUseLocalAppSource() {
 		return useLocalAppSource;
 	}
 
-	public InputStream createInputStream(String fileName) throws FileNotFoundException {
-		return new FileInputStream(fileName);
+	public InputStream createInputStream(String fileName) throws IOException {
+	    if (!OpenshiftUIConstants.URL_VALIDATOR.isValid(fileName)) {
+	        return new FileInputStream(fileName);
+	    } else {
```

Comment:
2 problems here:
- this method is called on the UI thread
- it ignores Eclipse's proxy settings. You should look at using ECF instead